trieste/acquisition/function/greedy.py

```diff
@@ -351,3 +355,127 @@ def __call__(self, x: TensorType) -> TensorType:
         p = -5  # following experiments of :cite:`Alvi:2019`.
         penalization = ((pairwise_distances / (self._radius + self._scale)) ** p + 1) ** (1 / p)
         return tf.reduce_prod(penalization, axis=-1)
+
+
+class FantasizeAcquisitionFunction(GreedyAcquisitionFunctionBuilder):
+    r"""
+    Builder of the acquisition function maker for greedily collecting batches, following the
+    kriging believer heuristic.
+
+    TODO: add stuff
+
+    Kriging believer allows us to perform batch Bayesian optimization with a standard (non-batch)
+    acquisition function.
+    By iteratively building a batch of points though sequentially maximizing
+    this acquisition function,
+    TODO: add stuff
+    it provides diverse batches of candidate points.
+    """
+
+    def __init__(
+        self,
+        acquisition_function_builder: AcquisitionFunctionBuilder = None,
+        fantasize_method: str = "KB",
```

Comment:
we could have a class attribute with a dictionary of these methods and a class method that the user can call to list all available methods

Reply:
could you provide an example? not sure what you mean

Reply:
check _REGISTRY in ModelRegistry in config.py as an example, we populate it with key-value pairs that act as available methods - names (keys) and any internal representation (values), then its easier to check whether correct method name is used, and we can have a static method that lists available methods (keys) for the user (e.g. get_registered_models in ModelRegistry)

for enum see https://www.tutorialspoint.com/enum-in-python, it would be another object, a bit unnecessary